networking/p2p/src/test/java/tech/pegasys/teku/networking/p2p/network/config/NetworkConfigTest.java

```diff
@@ -57,6 +60,36 @@ void getAdvertisedIp_shouldResolveLocalhostIpWhenInterfaceIpIsAnyLocalIpv6() {
     assertThat(result).isNotEqualTo("0.0.0.0");
   }
 
+  @Test
+  void shouldThrowExceptionIfInvalidFileName(@TempDir Path tempDir) {
+    final NetworkConfig config =
+        NetworkConfig.builder()
+            .advertisedIp(advertisedIp)
+            .networkInterface(listenIp)
+            .privateKeyFile(tempDir + "/invalid file name!!\0")
+            .build();
+
+    assertThat(config.getPrivateKeySource()).isPresent();
+    assertThatThrownBy(() -> config.getPrivateKeySource().get().getOrGeneratePrivateKeyBytes())
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessageContaining("Not able to create or retrieve p2p private key file -");
+  }
+
+  @Test
+  void shouldThrowExceptionIfProvideDirectory(@TempDir Path tempDir) {
+    final NetworkConfig config =
+        NetworkConfig.builder()
+            .advertisedIp(advertisedIp)
+            .networkInterface(listenIp)
+            .privateKeyFile(tempDir.toString()) // directory provided instead of file
+            .build();
+
+    assertThat(config.getPrivateKeySource()).isPresent();
+    assertThatThrownBy(() -> config.getPrivateKeySource().get().getOrGeneratePrivateKeyBytes())
+        .isInstanceOf(RuntimeException.class)
+        .hasMessageContaining("p2p private key file not found -");
+  }
```

Comment:
These tests are all about testing `FilePrivateKeySource`'s failure behaviours and should be on `FilePrivateKeySourceTest`.

Reply:
I believe what makes sense to test on `NetworkConfigTest`  is that we are correctly creating a `PrivateKeySource` when we use the builder with `privateKeyFile`